src/mrpro/data/_KData.py

```diff
@@ -171,3 +171,89 @@ def reshape_acq_data(data):
+            The dtype of the trajectory (torch.float32 or torch.float64) is then inferred from this.
+        non_blocking
+            If True and the source is in pinned memory, the copy will be asynchronous with respect to the host.
+            Otherwise, the argument has no effect.
+        copy
+            If True a new Tensor is created even when the Tensor already matches the desired conversion.
+        memory_format
+            The desired memory format of returned Tensor.
+        """
+        # Only complex64 and complex128 supported for kdata.
+        # This will then lead to a trajectory of float32 and float64, respectively.
+        if dtype is None:
+            dtype_traj = None
+        elif dtype == torch.complex64:
+            dtype_traj = torch.float32
+        elif dtype == torch.complex128:
+            dtype_traj = torch.float64
+        else:
+            raise ValueError(f'dtype {dtype} not supported. Only torch.complex64 and torch.complex128 is supported.')
+
+        return KData(
+            header=self.header,
+            data=self.data.to(
+                device=device, dtype=dtype, non_blocking=non_blocking, copy=copy, memory_format=memory_format
+            ),  # type: ignore [call-overload]
+            traj=self.traj.to(
+                device=device, dtype=dtype_traj, non_blocking=non_blocking, copy=copy, memory_format=memory_format
+            ),
+        )
+
+    def cuda(
+        self,
+        device: torch.device | None = None,
+        non_blocking: bool = False,
+        memory_format: torch.memory_format = torch.preserve_format,
+    ) -> KData:
+        """Create copy of object with trajectory and data in CUDA memory.
+
+        Parameters
+        ----------
+        device
+            The destination GPU device. Defaults to the current CUDA device.
+        non_blocking
+            If True and the source is in pinned memory, the copy will be asynchronous with respect to the host.
+            Otherwise, the argument has no effect.
+        memory_format
+            The desired memory format of returned Tensor.
+        """
+        return KData(
+            header=self.header,
+            data=self.data.cuda(
+                device=device, non_blocking=non_blocking, memory_format=memory_format
+            ),  # type: ignore [call-arg]
+            traj=self.traj.cuda(device=device, non_blocking=non_blocking, memory_format=memory_format),
+        )
+
+    def cpu(self, memory_format: torch.memory_format = torch.preserve_format) -> KData:
+        """Create copy of object in CPU memory.
+
+        Parameters
+        ----------
+        memory_format
+            The desired memory format of returned Tensor.
+        """
+        return KData(
+            header=self.header,
+            data=self.data.cpu(memory_format=memory_format),  # type: ignore [call-arg]
+            traj=self.traj.cpu(memory_format=memory_format),
```

Comment:
why is # type: ignore [call-overload] only required for one of the tensors?
Is mypy not checking the other one ?

Reply:
..this would mean that there is any issue with our typing somewhere else.
.data and .traj should behave the same wrt type hints.

Reply:
no .data is of type Tensor but .traj is of type Trajectory

Reply:
ah, thanks, got it